appbars/src/main/java/com/decathlon/vitamin/compose/appbars/bottomnavigations/VitaminBottomNavigations.kt

```diff
@@ -62,4 +67,48 @@ object VitaminBottomNavigations {
             }
         }
     }
+
+    /**
+     * The bottom navigation allow the user to switch between destinations in an app.
+     * @param actions The [SelectedActionItem] actions of your bottom navigation.
+     * [SelectedActionItem] define the look and the event associated to an item in the bottom navigation
+     * @param modifier The [Modifier] to be applied to this BottomNavigation
+     * @param colors The colors of the background and the content elements in selected and unselected mode
+     */
+    @Composable
+    fun PrimaryImmutable(
```

Comment:
I'm not a big fan of putting the suffix `Immutable` for every method which have to be used instead, IMO only the deprecated method should be suffixed and the right method should keep the "right" name. Because it's a library, it's impossible to do this without breaking everything which depends on. So I think there is an other solution:
* Deprecate the object `VitaminBottomNavigations`
* Create a new object with the same name `VitaminBottomNavigations` into a different package (`com.decathlon.vitamin.compose.appbars.bottomnavigations.immutable` for instance)
* Put the new methods into this object without the `Immutable` suffix and let the Deprecated methods into the old package name
* Add `replaceWith = ReplaceWith("com.decathlon.vitamin.compose.appbars.bottomnavigations.immutable.VitaminBottomNavigations")` to the `@Deprecated` annotation

By doing this, the user only have to change imports, the usage of the `ReplaceWith` feature is trivial and the method naming remains "right".

What do you think about this ?

Reply:
The depreciation is an hard topic because you can't predict all evolutions in the future... Your proposal is interesting but with some new pain points:

1. If you have other components inside the object class which aren't deprecated, you force consumers to use the new object class even if it isn't necessary.
2. Add a suffix in the name of the component or duplicate the object class somewhere else only moves the problem.
3. We need to find a depreciation strategy we can use in the future for other use cases. Move the new object class inside `immutable` package isn't perennial in time.

I have a new proposition: For all components, we create a package `components` (or `modifiers`, cf. comment about skeletons). Inside this new package, we create multiple new packages `v1`, `v2`, etc. according to our depreciation needs.

And we put all our components inside these packages but as function extension. So you can have the same name than the previous version, you keep the auto completion experience inside Android Studio and your import (in consumer side) drive the component you want to use.

```kotlin
// components.v1
@Deprecated(
    message = "Use Primary variant with expanded parameter typed with State and no more MutableState."
)
@Composable
fun VitaminTopBars.Primary(
    // parameters
) {
    // stuff
}
```

```kotlin
// components.v2
@Composable
fun VitaminTopBars.Primary(
    // parameters
) {
    // stuff
}
```

```kotlin
// usage
import com.decathlon.vitamin.compose.appbars.topbars.components.v2.Primary

VitaminTopBars.Primary(/* stuff */)
```